tools/cli/src/mentat_cli/command_parser.rs

```diff
@@ -62,18 +65,44 @@ pub fn command(s: &str) -> Command {
                     });
 
     let close_parser = string(CLOSE_COMMAND)
-                    .and(skip_many(space()))
+                    .and(skip_spaces.clone())
                     .map( |_| Command::Close );
 
-    skip_many(space())
+    let opening_brace_parser = try(string("["))
+                    .or(try(string("{")));
+    
+    let edn_arg_parser = skip_spaces.clone()
+                    .and(opening_brace_parser.clone()
+                    .and(many::<Vec<_>, _>(try(any()))));
+
+    let query_parser = string(QUERY_COMMAND)
+                    .and(edn_arg_parser.clone())
```

Comment:
See my other commit with `.skip` and `.then` for avoiding these indexing operations.